docs/agent/hostname-resolution.md

```diff
@@ -0,0 +1,26 @@
+# Difference in hostname resolution between Agent v5 and Agent v6 (<v6.4)
+
+In some cases, it is possible to see a difference in the hostname that’s reported by your Agent when upgrading from Agent v5 to Agent v6 (for versions < 6.4). 
+
+To resolve the system hostname the Agent 5 uses the `hostname -f` command while the Agent v6 (for versions < 6.4) uses the Golang API `os.Hostname()`. 
+
+The most common issue this can create is to change your hostname from a Fully-Qualified Domain Name (FQDN, ex. sub.domain.tld) to a short hostname (ex. sub). 
+
+Starting from the Agent v6.3 a configuration flag called `hostname_fqdn` has been introduced that allows the Agent v6 to have the same behavior as Agent v5. This flag is disabled by default on version 6.3 and enabled by default in version 6.4.
+
```

Comment:
Let's add a sub-title here: `Determine if you're affected` (or something like that, not sure about the exact title)